src/CommunityPatch/Patches/EquipDraggedNonCivilianItemPatch.cs

```diff
@@ -0,0 +1,296 @@
+        return false;
+      }
+
+      if (!IsTargetPatchable(UpdateCurrentCharacterIfPossibleMethodInfo, UpdateCurrentCharacterIfPossibleHashes)) {
+        return false;
+      }
+      
+      if (!IsTargetPatchable(InitializeInventoryMethodInfo, InitializeInventoryHashes)) {
+        return false;
+      }
+      
+      if (!IsTargetPatchable(AfterTransferMethodInfo, AfterTransferHashes)) {
+        return false;
+      }
+
+      return base.IsApplicable(game);
+    }
+
+    private static void MapLeftAndRightInventory(SPInventoryVM spInventoryVm, Action<SPItemVM> itemMapper) {
+      var rightItemListVm = (MBBindingList<SPItemVM>) RightItemListVmFieldInfo.GetValue(spInventoryVm);
+      var leftItemListVm = (MBBindingList<SPItemVM>) LeftItemListVmFieldInfo.GetValue(spInventoryVm);
+
+      if (rightItemListVm == null || leftItemListVm == null) {
+        return;
+      }
+      
+      leftItemListVm.ApplyActionOnAllItems(itemMapper);
+      rightItemListVm.ApplyActionOnAllItems(itemMapper);
+    }
+
+    private static void NonCivilianPerkItemsEquipableState(SPInventoryVM spInventoryVm, Func<SPItemVM, bool> isItemApplicableToPerk, bool state) {
+
+      Action<SPItemVM> makeCivilianItemUnequipable = item => {
+        var isCivilian = item?.ItemRosterElement.EquipmentElement.Item?.IsCivilian ?? false;
+        if (isItemApplicableToPerk(item) && !isCivilian) {
+          SetIsCivilianItemMethodInfo.Invoke(item, new object[] { state });
+        }
+      };
+      
+      MapLeftAndRightInventory(spInventoryVm, makeCivilianItemUnequipable);
+    }
+
+    private static void MakeNonCivilianPerkItemsUnequipable(SPInventoryVM spInventoryVm, Func<SPItemVM, bool> isItemApplicableToPerk)
+      => NonCivilianPerkItemsEquipableState(spInventoryVm, isItemApplicableToPerk, false);
+
+    private static void MakeNonCivilianPerkItemsEquipable(SPInventoryVM spInventoryVm, Func<SPItemVM, bool> isItemApplicableToPerk)
+      => NonCivilianPerkItemsEquipableState(spInventoryVm, isItemApplicableToPerk, true);
+    
+    private static bool CharacterHasPerk(CharacterObject characterObject) {
+      return characterObject != null && characterObject.GetPerkValue(ActivePatch.Perk);
+    }
+
+    private static void FlipEquipableStateOfNonCivilianPerkItemsIfCharacterHasPerkOrNot(SPInventoryVM spInventoryVm, CharacterObject previousCharacter, Func<SPItemVM, bool> isItemApplicableToPerk) {
+      if (CharacterHasPerk(previousCharacter)) {
+        MakeNonCivilianPerkItemsEquipable(spInventoryVm, isItemApplicableToPerk);
+      }
+      else {
+        MakeNonCivilianPerkItemsUnequipable(spInventoryVm, isItemApplicableToPerk);
+      }
+    }
+    
+    protected static void FlipEquipableStateOfNonCivilianPerkItemsIfDisplayedCharacterHasPerkOrNot(SPInventoryVM spInventoryVm, Func<SPItemVM, bool> isItemApplicableToPerk) {
+      var currentCharacter = (CharacterObject)CurrentCharacterFieldInfo.GetValue(spInventoryVm);
```

Comment:
Try to use BuildGetter

Reply:
By trying out the following :

```cs
    internal static readonly FieldInfo CurrentCharacterFieldInfo = typeof(SPInventoryVM)
      .GetField("_currentCharacter", NonPublic | Instance | DeclaredOnly);
      
    internal static readonly Func<SPInventoryVM, CharacterObject> CurrentCharacterGetter = CurrentCharacterFieldInfo
      .BuildGetter<SPInventoryVM, CharacterObject>();
```

CurrentCharacterGetter retuns null during runtime. So going to use AccessTools.FieldRefAccess method instead.